pkg/gatewayserver/io/grpc/grpc.go

```diff
@@ -47,14 +47,14 @@ func (s *impl) LinkGateway(link ttnpb.GtwGs_LinkGatewayServer) (err error) {
 	ids := ttnpb.GatewayIdentifiers{
 		GatewayID: rpcmetadata.FromIncomingContext(ctx).ID,
 	}
-	if err = ids.ValidateContext(ctx); err != nil {
+	ctx, ids, err = s.server.FillGatewayContext(ctx, ids)
+	if err != nil {
 		return
 	}
-	if err = rights.RequireGateway(ctx, ids, ttnpb.RIGHT_GATEWAY_LINK); err != nil {
+	if err = ids.ValidateContext(ctx); err != nil {
 		return
 	}
-	ctx, ids, err = s.server.FillGatewayContext(ctx, ids)
-	if err != nil {
+	if err = rights.RequireGateway(ctx, ids, ttnpb.RIGHT_GATEWAY_LINK); err != nil {
```

Comment:
This actually resolves a never-tested panic; we were requiring gateway rights before filling the context (with the rights fetcher). With this PR we test this correctly.